### PR TITLE
Performance and other fixes

### PR DESCRIPTION
**Pull request by DanWin**
This fixes some performance issues, a memory leak and an incomplete copy-constructor I found using cppcheck.

### CODE REVIEW

File: plugins/LadspaEffect/calf/src/synth.cpp [hunk 1]
```diff
@@ -25,7 +25,7 @@ using namespace std;
 
 void basic_synth::kill_note(int note, int vel, bool just_one)
 {
-    for (list<dsp::voice *>::iterator it = active_voices.begin(); it != active_voices.end(); it++) {
```
Comment (tresf) [inline plugins/LadspaEffect/calf/src/synth.cpp]: Hmm... @Lukas-W @diizy Are these old calf plugins our problem now?  I see some tweaks are being made, but upstream has diverged too much in this case, right?

**Comment by tresf** [comment]
I spent some time researching these optimizations i.e. `++i` versus `i++`, use of addresses for `QString &` rather than allowing QT to do implicit sharing (which costs a tiny bit of CPU/Memory).

The only ones I'm not sure of is `PlayHandle.h`'s uninitialized member variables, but we'll see. :smile:

**Comment by badosu** [comment]
I don't understand, why does `++i` vs `i++` make a difference? This statement is so simple that should be really straightforward to be optimized by the compiler.

**Comment by tresf** [comment]
@badosu you are correct, but if it makes cppcheck happier, it is probably worth the commit to prevent resurfacing down the road. *shrug* :smiling_imp: 

http://stackoverflow.com/questions/24886/is-there-a-performance-difference-between-i-and-i-in-c

**Comment by DanWin** [comment]
@badosu @tresf  There is a difference for objects in iterations. The compiler might not be able to optimize them, as it doesn't know, whether it++ was intended or could be optimized to ++it. For simple int i = 0; … i++; the compiler does optimize the code, because it's more obvious.

http://www.idinews.com/peeves/prefixIncr.html

**Comment by DanWin** [comment]
@tresf The PlayHandle.h member variables got added in a few commits after the copy constructor got added, but they were probably forgotten there.

**Comment by tresf** [comment]
Thanks for the explanations.

File: plugins/LadspaEffect/calf/src/synth.cpp (hunk 1 above)
Comment (diizy) [inline plugins/LadspaEffect/calf/src/synth.cpp]: On 01/09/2015 09:38 PM, Tres Finocchiaro wrote:
>
> In plugins/LadspaEffect/calf/src/synth.cpp
> <https://github.com/LMMS/lmms/pull/1585#discussion-diff-22737757>:
>
> > @@ -25,7 +25,7 @@ using namespace std;
> >  
> >  void basic_synth::kill_note(int note, int vel, bool just_one)
> >  {
> > -    for (list<dsp::voice *>::iterator it = active_voices.begin(); it != active_voices.end(); it++) {
>
> Hmm... @Lukas-W <https://github.com/Lukas-W> @diizy
> <https://github.com/diizy> Are these old calf plugins our problem now?
> I see some tweaks are being made, but upstream has diverged too much
> in this case, right?
>

That looks like Calf synth plugin code which doesn't work in any LMMS
plugins, probably DSSI/LV2 stuff. Touching it at all is pointless.

Comment (tresf) [inline plugins/LadspaEffect/calf/src/synth.cpp]: Thanks for the insight.  No harm no foul I suppose.

**Comment by badosu** [comment]
@DanWin Thanks for the explanation!

**Pull request merged**